### DISCUSSION TITLE
For anyone with compiling issues

### DISCUSSION THREAD

**Discussion by ghost** [KadeArchive/Kade-Engine#2834] [General]
I know many people out there are having many problems with compiling any version of Kade Engine, so I will tell you some of the most common errors and how to fix them.

### Game Crashes Right After Caching ###
This happens with every outdated version of Kade Engine. (I think) For versions 1.7.1 and earlier, the game would receive an error immediately after opening the title state and would crash. More explained in #2291. To fix this, open TitleState.hx and delete this line.
> https://github.com/KadeDev/Kade-Engine/blob/e0c5c578ad228284cc68fe10067aabafef9111dd/source/TitleState.hx#L57

### CMD Line Outputs extension-webm Error ###
Its most likely you didn't put in the git commands correctly.
Type this in the cmd line
```haxelib git extension-webm```
THEN you type in the link after. Don't put the link on the same line as the haxelib git command.
```https://github.com/KadeDev/extension-webm```
Then rebuild the extension-webm by typing this.
```lime rebuild extension-webm windows```
If your using another device, then replace "windows" with the type of device your using.

### WEBM Player Error in PlayState ###
Most likely the version of git that your using. Turns out the version to use is also really strict with Git and not Haxe.
Download Git 2.31.1 instead and delete the old git. Then redo the cmd commands and try again.
Download it with this link. WARNING! IT WILL AUTOMATICALLY DOWNLOAD! https://mirrors.edge.kernel.org/pub/software/scm/git/git-2.31.1.tar.gz

### The Game Was Already Built/I Don't Have Access To The HX Files ###
These kinds of stupid mistakes always make me laugh so hard!
You totally downloaded the built version and not the source version. Sucks to suck man.
Just do Code > Code > Download ZIP

### Common Things People Forget ###
1. KADE ENGINE IS COMPILED DIFFERENTLY THAN VANILLA FNF!
For Kade Engine, you need to follow this link and not ninjamuffin99's link.
Kade Tutorial: https://github.com/KadeDev/Kade-Engine/blob/stable/docs/building.md
Vanilla Tutorial: https://github.com/ninjamuffin99/Funkin/blob/master/README.md
2. TO RUN THE LIME TEST WINDOWS COMMAND, THE DIRECTORY HAS TO BE THE GAME FOLDER!
Probably something nobody ever forgets, but I'm gonna say it anyway.
The game has to be compiled with the command prompt's directory set to the folder.
Open the folder of the engine and make sure you can see the project.xml file.
Then click the box showing the folder you are in and type cmd. (or if you want to use Windows PowerShell, type powershell)
Now you can use the command.

That's all I could think of for now. I hope this helps! :)

**Comment by qq735-personal** [comment]
followed all instructions and it says lime doesnt exist when i put in `lime build windows`

**Reply by ghost** [reply]
Try running these two commands again.
```haxelib run lime setup```
```haxelib run lime setup flixel```

**Comment by Preglas** [comment]
I can't run the executable i compiled for linux, it says that i need an app to run it.
Also yes i have in Allow for executing checked
I'm using Linux Mint newest version
![Zrzut ekranu z 2022-02-09 19-30-11](https://user-images.githubusercontent.com/97025494/153266446-ae7521d5-a6cf-4bbb-bca1-b7805d7abf44.png)
